### DISCUSSION TITLE
URL rewriting with Deno's `serveDir`

### DISCUSSION THREAD

**Discussion by youk** [honojs/hono#1740] [Q&A]
Is it so that currently it is not possible to rewrite URLs of the files served via [serveDir](https://deno.land/std/http/file_server.ts?s=serveDir)? The way it is implemented, the only option I can think of is to change the URL of incoming `Request` object. The problem is that `url` is read-only in `HonoRequest` (I assume that is by design?). I can tweak it by replacing the `url` getter, but it is merely a hack.

**Comment by EdamAme-x** [comment]
Since `serveDir` is a standard collection of Deno, it has nothing to do with `hono`...?

Instead, the following functions are available :)
https://hono.dev/getting-started/deno#rewriterequestpath

**Reply by youk** [reply]
> Since serveDir is a standard collection of Deno, it has nothing to do with hono...?

It has _something_ to do with Deno as this is a matter of integration. The existence of [Adapter Helper](https://hono.dev/helpers/adapter) clearly shows this.

> https://hono.dev/getting-started/deno#rewriterequestpath

Thanks, but I need to serve dynamic resources.
